# Erste Schritte

### Addition, Subtraktion und Multiplikation

Es folgen die Eingabe, die Verarbeitung und die Ausgabe einiger einfacher Berechungen:

In [3]:
41 + 7.5

21

In [4]:
12 - 8

4

In [5]:
7 * 3

21

Der Reihe nach werden die Operatoren + (Addition), - (Subtraktion) und * (Multiplikation) genutzt. Sie können sowohl mit ganzen Zahlen rechen als auch mit Zahlen, die Nachkommastellen besitzen. Die Nachkommastellen müssen
mithilfe eines Dezimalpunkts abgetrennt werden.

### Division, Ganzzahldivision und Modulo

Der Operator / dient der mathematischen Division:

In [6]:
22 / 8

2.75

In [7]:
22 / -8

-2.75

Mithilfe des Operators // wird eine Ganzzahldivision durchgeführt:

In [ ]:
22 // 8

In [ ]:
22 // -8

In [ ]:
7 // 2.5

Bei einer Ganzzahldivision mithilfe des Operators // wird zunächst das Ergebnis der mathematischen Division berechnet. Anschließend wird die nächstkleinere ganze Zahl ermittelt: Aus 22 / 8 = 2.75 wird 2, aus 22 / -8 = -2.75
wird -3. Handlet es sich bei midestens einem der Operanden um eine Zahl mit Nachkommastellen, erscheint auch das Ergebnis als solche Zahl: Aus 7 / 2.5 = 2.8 wird 2.0.

Der Modulo-Operator % berechnet den Rest einer Ganzzahldivision:

In [ ]:
22 % 8

In [8]:
22.5 % 8.5

5.5

Die Ganzzahldivision 22 // 8 ergibt "2 Rest 6". Der Modulo-Operator liefert den Rest 6. Die Ganzzahldivision 22.5 // 8.5 ergibt "2.0 Rest 5.5". Der Modulo-Operator liefert den Rest 5.5.